table/table.go

```diff
@@ -452,3 +463,238 @@ func (table *Table) Print() (str string) {
 	}
 	return
 }
+
+func InitFromConfig(config cfg.Config) (*Table, error) {
+	table := New(config.Spool_dir)
+
+	err := table.InitCmd(config)
+	if err != nil {
+		return table, err
+	}
+
+	err = table.InitBlacklist(config)
+	if err != nil {
+		return table, err
+	}
+
+	err = table.InitAggregation(config)
+	if err != nil {
+		return table, err
+	}
+
+	err = table.InitRewrite(config)
+	if err != nil {
+		return table, err
+	}
+
+	err = table.InitRoutes(config)
+	if err != nil {
+		return table, err
+	}
+
+	return table, nil
+}
+
+func (table *Table) InitCmd(config cfg.Config) error {
+	for i, cmd := range config.Init {
+		log.Notice("applying: %s", cmd)
+		err := imperatives.Apply(table, cmd)
+		if err != nil {
+			log.Error(err.Error())
+			return fmt.Errorf("could not apply init cmd #%d", i+1)
+		}
+	}
+
+	return nil
+}
+
+func (table *Table) InitBlacklist(config cfg.Config) error {
+	for i, entry := range config.BlackList {
+		parts := strings.SplitN(entry, " ", 2)
+		if len(parts) < 2 {
+			return fmt.Errorf("invalid blacklist cmd #%d", i+1)
+		}
+
+		prefix_pat := ""
+		sub_pat := ""
+		regex_pat := ""
```

Comment:
golang variable naming style is regexPatt, subPatt etc. I know this codebase is already a mismash of underscoring vs camelCasing naming, but i hope to some day address it. you don't have to change it now. we can do it later. BTW which editor do use?  vim-go has a nice integration with https://godoc.org/golang.org/x/tools/cmd/gorename which makes this sort of thing less painful.